phy/cluster/manual/clustering.py

```diff
@@ -113,56 +165,83 @@ def __init__(self, spike_clusters):
         self._spike_clusters_base = self._spike_clusters.copy()
 
     def reset(self):
+        """Reset the clustering to the original clustering.
+
+        All changes are lost.
+
+        """
+        self._undo_stack.clear()
         self._spike_clusters = self._spike_clusters_base
         self._update_all_spikes_per_cluster()
 
     @property
     def spike_clusters(self):
```

Comment:
Not really. For example, the undo stack is fundamentally based on this array. Also, you need this array when you want to compute all pairwise CCGs in a set of neurons. Finally, this is what is stored in the file format.